app/plugin.go

```diff
@@ -37,6 +37,35 @@ var prepackagedPlugins map[string]func(string) ([]byte, error) = map[string]func
 	"zoom": zoom.Asset,
 }
 
+func (a *App) notifyPluginStatusesChanged() error {
+	result := <-a.Srv.Store.Plugin().GetPluginStatuses()
+	if result.Err != nil {
+		return result.Err
+	}
+
+	pluginStatuses := result.Data.([]*model.PluginStatus)
+
+	// Notify any system admins.
+	message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_PLUGIN_STATUSES_CHANGED, "", "", "", nil)
+	message.Add("plugin_statuses", pluginStatuses)
+	message.Broadcast.ContainsSensitiveData = true
+	a.Publish(message)
+
+	return nil
+}
+
+func (a *App) setPluginStatusState(id string, state int) error {
+	if result := <-a.Srv.Store.Plugin().UpdatePluginStatusState(&model.PluginStatus{
+		ClusterDiscoveryId: a.GetClusterId(),
+		PluginId:           id,
+		State:              state,
+	}); result.Err != nil {
+		return result.Err
+	}
+
+	return a.notifyPluginStatusesChanged()
```

Comment:
This effectively pushes down the entire plugin statuses structure to all listening admin clients. On the one hand, this seems like overkill, except it simplifies certain flows, and in the grand scheme of things (num plugins and num admins), seems small enough.